web/modules/custom/collection/src/Plugin/Block/CollectionContentBlock.php

```diff
@@ -87,7 +87,9 @@ public static function create(ContainerInterface $container, array $configuratio
    */
   public function build() {
     if (empty($this->collection)) {
-      throw new \Exception('The "Collection content" block can only be shown on collection pages.');
+      // If this is not a collection page, return an empty array so that the
+      // block is not rendered.
+      return [];
```

Comment:
The reason is the very context we are implementing in this ticket.
If the context is available, then the condition of "shown to collection pages" is becoming true. And while with the context present, you can consider this page as a collection pages, still, you would not want the block with the content of the collection to be shown to a (e.g.) news page that belong to this collection.

Also, IMHO, if you consider the usability of this block, it is meant to show some related content. Misplacing it to a page should not break the page throwing an exception because this change is regarding actually the visibility of the block. returning an empty array simple tells drupal to not render it.